src/screens/WalletConnect/CallRequest/TransactionRequestContent.js

```diff
@@ -158,18 +166,31 @@ const useTransactionFee = (request: WalletConnectCallRequest) => {
 
 const useViewData = (request: WalletConnectCallRequest) => {
   const { t } = useTranslation();
+  const activeAccount = useActiveAccount();
+  const isArchanovaAccountDeployed = useRootSelector(isArchanovaAccountDeployedSelector);
+
+  const isArchanovaAccountActive = isArchanovaAccount(activeAccount);
 
   let errorMessage = null;
   const chain = chainFromChainId[request.chainId];
   if (!chain) {
     errorMessage = t('error.walletConnect.cannotDetermineEthereumChain');
   }
 
-  const isDeployedOnChain = useRootSelector(isDeployedOnChainSelector);
+  if (isArchanovaAccountActive && chain !== CHAIN.ETHEREUM) {
+    errorMessage = t('error.walletConnect.activeAccountDoesNotSupportSelectedChain');
+  }
+
   const estimationErrorMessage = useRootSelector((root) => root.transactionEstimate.errorMessage);
 
+  /**
+   * Archanova account needs to be deployed regardless of the action,
+   * Etherspot account must to be deployed for signature requests only
```

Comment:
```suggestion
   * Archanova account needs to be deployed for all call requests.
   * Etherspot account doesn't need to be deployed for transaction requests.
```

Reply:
This is signature request so I would leave signature keyword here:

```
 /**
   * Archanova account needs to be deployed for all types call requests.
   * Etherspot account needs to be deployed for signature type call requests only.
   */
```